app/src/main/java/de/danoeh/antennapod/adapter/AllEpisodesRecycleAdapter.java

```diff
@@ -23,29 +26,38 @@
 import java.lang.ref.WeakReference;
 
 import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
 import de.danoeh.antennapod.core.feed.FeedItem;
 import de.danoeh.antennapod.core.feed.FeedMedia;
 import de.danoeh.antennapod.core.glide.ApGlideSettings;
+import de.danoeh.antennapod.core.storage.DownloadRequestException;
 import de.danoeh.antennapod.core.storage.DownloadRequester;
 import de.danoeh.antennapod.core.util.Converter;
 import de.danoeh.antennapod.core.util.NetworkUtils;
+import de.danoeh.antennapod.fragment.ItemFragment;
+import de.danoeh.antennapod.menuhandler.FeedItemMenuHandler;
 
 /**
  * List adapter for the list of new episodes
  */
-public class AllEpisodesListAdapter extends BaseAdapter {
+public class AllEpisodesRecycleAdapter extends RecyclerView.Adapter<AllEpisodesRecycleAdapter.Holder> {
 
-    private static final String TAG = AllEpisodesListAdapter.class.getSimpleName();
+    private static final String TAG = AllEpisodesRecycleAdapter.class.getSimpleName();
 
     private final Context context;
     private final ItemAccess itemAccess;
     private final ActionButtonCallback actionButtonCallback;
     private final ActionButtonUtils actionButtonUtils;
     private final boolean showOnlyNewEpisodes;
+    private final MainActivity mainActivity;
```

Comment:
Should not really cause an activity leak (as long as we set the adapter in the activity to null), but I could sleep better if this was a WeakReference